src/components/TodoList.tsx

```diff
@@ -0,0 +1,55 @@
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+interface TodoListProps {
+  todos: Todo[],
+  filter: Filter,
+}
+
+export const TodoList: React.FC<TodoListProps> = ({ todos, filter }) => {
+  const filteredTodos = todos.filter(todo => {
+    switch (filter) {
+      case Filter.All:
+        return true;
+      case Filter.Active:
+        return !todo.completed;
+      case Filter.Completed:
+        return todo.completed;
+      default:
+        throw new Error('wrong filter selected');
+    }
+  });
```

Comment:
Good defensive programming practice, even if the enum currently has only three hardcoded types.
 Don't forget put empty lines after returns for better readability.